tests/mailinglist.rs

```diff
@@ -0,0 +1,332 @@
+#![feature(async_closure)]
+
+extern crate diesel;
+extern crate rocket;
+
+use lettre::message::Mailbox;
+use rocket::http::{Accept, ContentType, Status};
+
+use zauth::config::Config;
+use zauth::mailer::Mailer;
+use zauth::models::mail::NewMail;
+use zauth::models::user::*;
+use zauth::DbConn;
+
+mod common;
+
+const TEST_USERS: [(&'static str, UserState, bool); 10] = [
+	("valid0", UserState::Active, true),
+	("valid1", UserState::Active, true),
+	("pending_approval0", UserState::PendingApproval, true),
+	("pending_approval1", UserState::PendingApproval, true),
+	("pending_email0", UserState::PendingMailConfirmation, true),
+	("pending_email1", UserState::PendingMailConfirmation, true),
+	("disabled0", UserState::Disabled, true),
+	("disabled1", UserState::Disabled, true),
+	("unsubbed0", UserState::Disabled, false),
+	("unsubbed01", UserState::Disabled, false),
+];
+
+async fn setup_test_users(db: &DbConn) {
+	for test_user in TEST_USERS {
+		let mut created_user = User::create(
+			NewUser {
+				username:    test_user.0.to_string(),
+				password:    format!(
+					"{}verylongandsecurepassword",
+					test_user.0.to_string()
+				),
+				full_name:   test_user.0.to_string(),
+				email:       format!("{}@example.com", test_user.0.to_string()),
+				ssh_key:     None,
+				not_a_robot: true,
+			},
+			common::BCRYPT_COST,
+			&db,
+		)
+		.await
+		.unwrap();
+
+		created_user.state = test_user.1;
+		created_user.subscribed_to_mailing_list = test_user.2;
+
+		created_user.update(&db).await.unwrap();
+	}
+}
+
+/// Check if the correct users are selected as being validly subscribed to the
+/// mailing list and if mailing list emails have them in bcc properly
+#[rocket::async_test]
+async fn get_valid_subscribed_users() {
+	common::as_visitor(async move |_client, db| {
+		setup_test_users(&db).await;
+
+		let subscribed_users = User::find_subscribed(&db).await.unwrap();
+
+		let subscribed_usernames: Vec<String> = subscribed_users
+			.iter()
+			.map(|u| u.username.clone())
+			.collect();
+
+		let bcc: Vec<Mailbox> = subscribed_users
+			.iter()
+			.map(|u| Mailbox::try_from(u).unwrap())
+			.collect();
+
+		let test_cfg = Config {
+			admin_email: "".to_string(),
+			user_session_seconds: 0,
+			client_session_seconds: 0,
+			authorization_token_seconds: 0,
+			email_confirmation_token_seconds: 0,
+			secure_token_length: 0,
+			bcrypt_cost: 0,
+			base_url: "".to_string(),
+			mail_queue_size: 5,
+			mail_queue_wait_seconds: 5,
+			mail_from: "test@example.com".to_string(),
+			mail_server: "stub".to_string(),
+			maximum_pending_users: 0,
+			webhook_url: None,
+		};
```

Comment:
```suggestion
		let test_cfg = common::config();
```
This removes some boilerplate. Is there a specific config value you want to set?

Reply:
github didn't let me merge the suggestion smh, fixed in (130582595f3e59a78d04e275b52cf40ee18c987d)